### ISSUE TITLE
[REGRESSION] Qemu driver is unable to connect to /opt/homebrew/var/run/socket_vmnet after switch from 1.31.2 to 1.32.0

### ISSUE THREAD

**Issue by DuMaM** [kubernetes/minikube#17615]
### What Happened?

# Error 

Since upgrade to 1.32.0 I'm unable to connect to socket_vmnnet.
ERROR: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Permission denied

# Env Setup 

## Mac M1
```sw_vers
ProductName:            macOS
ProductVersion:         13.6.1
BuildVersion:           22G313
```

## minikube

``` sh
 ~/W/w/web-docker-dev   feature..rnetes ±  brew info minikube
==> minikube: stable 1.32.0 (bottled), HEAD
Run a Kubernetes cluster locally
https://minikube.sigs.k8s.io/
/opt/homebrew/Cellar/minikube/1.32.0 (9 files, 87.7MB) *
  Poured from bottle using the formulae.brew.sh API on 2023-11-13 at 16:45:31
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/m/minikube.rb
License: Apache-2.0
==> Dependencies
Build: go ✔, go-bindata ✘
Required: kubernetes-cli ✔
==> Options
--HEAD
        Install HEAD version
==> Caveats
fish completions have been installed to:
  /opt/homebrew/share/fish/vendor_completions.d
==> Analytics
install: 12,313 (30 days), 36,286 (90 days), 94,775 (365 days)
install-on-request: 12,306 (30 days), 36,272 (90 days), 94,729 (365 days)
build-error: 1 (30 days)
```

## qemu

``` sh
 ~/W/w/web-docker-dev   feature..rnetes ±  brew info qemu
==> qemu: stable 8.1.2 (bottled), HEAD
Emulator for x86 and PowerPC
https://www.qemu.org/
/opt/homebrew/Cellar/qemu/8.1.2 (162 files, 534.1MB) *
  Poured from bottle using the formulae.brew.sh API on 2023-10-23 at 10:16:36
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/q/qemu.rb
License: GPL-2.0-only
==> Dependencies
Build: libtool ✔, meson ✘, ninja ✘, pkg-config ✔, spice-protocol ✔
Required: capstone ✔, dtc ✔, glib ✔, gnutls ✔, jpeg-turbo ✔, libpng ✔, libslirp ✔, libssh ✔, libusb ✔, lzo ✔, ncurses ✔, nettle ✔, pixman ✔, snappy ✔, vde ✔, zstd ✔
==> Options
--HEAD
        Install HEAD version
==> Analytics
install: 45,047 (30 days), 159,567 (90 days), 332,573 (365 days)
install-on-request: 34,222 (30 days), 122,544 (90 days), 255,055 (365 days)
build-error: 214 (30 days)
```

## socet_vmn

```
 ~/W/w/web-docker-dev   feature..rnetes ±  brew info socket_vmnet 
==> socket_vmnet: stable 1.1.3 (bottled), HEAD [keg-only]
Daemon to provide vmnet.framework support for rootless QEMU
https://github.com/lima-vm/socket_vmnet
/opt/homebrew/Cellar/socket_vmnet/1.1.3 (14 files, 121.3KB)
  Poured from bottle using the formulae.brew.sh API on 2023-11-13 at 16:54:48
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/s/socket_vmnet.rb
License: Apache-2.0
==> Requirements
Required: macOS ✔, macOS >= 10.15 (or Linux) ✔
==> Options
--HEAD
        Install HEAD version
==> Caveats
socket_vmnet requires root privileges so you will need to run
  `sudo /opt/homebrew/opt/socket_vmnet/socket_vmnet` or `sudo brew services start socket_vmnet`.
You should be certain that you trust any software you grant root privileges.

socket_vmnet is keg-only, which means it was not symlinked into /opt/homebrew,
because Homebrew's bin directory is often writable by a non-admin user.

If you need to have socket_vmnet first in your PATH, run:
  fish_add_path /opt/homebrew/opt/socket_vmnet/bin

To start socket_vmnet now and restart at startup:
  sudo brew services start socket_vmnet
Or, if you don't want/need a background service you can just run:
  /opt/homebrew/opt/socket_vmnet/bin/socket_vmnet --vmnet-gateway\=192.168.105.1 /opt/homebrew/var/run/socket_vmnet
==> Analytics
install: 607 (30 days), 1,655 (90 days), 2,917 (365 days)
install-on-request: 607 (30 days), 1,653 (90 days), 2,916 (365 days)
build-error: 0 (30 days)
 ~/W/w/web-docker-dev   feature..rnetes ±  
 ```

### Attach the log file

Logs

```
+ minikube start -p minikube --install-addons=true --driver=qemu2 --cni=cilium --network=socket_vmnet --disk-size=40g --cpus=6 --memory=8g --nodes=1 --mount-string /Users/nowakb/Work/webp:/web --mount --alsologtostderr -v=7
I1113 17:40:43.627308   11335 out.go:296] Setting OutFile to fd 1 ...
I1113 17:40:43.627469   11335 out.go:348] isatty.IsTerminal(1) = true
I1113 17:40:43.627472   11335 out.go:309] Setting ErrFile to fd 2...
I1113 17:40:43.627475   11335 out.go:348] isatty.IsTerminal(2) = true
I1113 17:40:43.627590   11335 root.go:338] Updating PATH: /Users/nowakb/.minikube/bin
I1113 17:40:43.629393   11335 out.go:303] Setting JSON to false
I1113 17:40:43.663448   11335 start.go:128] hostinfo: {"hostname":"WVLKPDWWKG","uptime":2393,"bootTime":1699891250,"procs":679,"os":"darwin","platform":"darwin","platformFamily":"Standalone Workstation","platformVersion":"13.6.1","kernelVersion":"22.6.0","kernelArch":"arm64","virtualizationSystem":"","virtualizationRole":"","hostId":"086bc434-f008-55ca-b6b1-3e51ce9da5c6"}
W1113 17:40:43.663560   11335 start.go:136] gopshost.Virtualization returned error: not implemented yet
I1113 17:40:43.668787   11335 out.go:177] 😄  [minikube] minikube v1.32.0 on Darwin 13.6.1 (arm64)
😄  [minikube] minikube v1.32.0 on Darwin 13.6.1 (arm64)
I1113 17:40:43.679136   11335 notify.go:220] Checking for updates...
I1113 17:40:43.679463   11335 config.go:182] Loaded profile config "minikube": Driver=qemu2, ContainerRuntime=docker, KubernetesVersion=v1.28.3
W1113 17:40:43.679494   11335 start.go:810] api.Load failed for minikube: filestore "minikube": Docker machine "minikube" does not exist. Use "docker-machine ls" to list machines. Use "docker-machine create" to add a new one.
I1113 17:40:43.679557   11335 driver.go:378] Setting default libvirt URI to qemu:///system
W1113 17:40:43.679573   11335 start.go:810] api.Load failed for minikube: filestore "minikube": Docker machine "minikube" does not exist. Use "docker-machine ls" to list machines. Use "docker-machine create" to add a new one.
I1113 17:40:43.684946   11335 out.go:177] ✨  Using the qemu2 driver based on existing profile
✨  Using the qemu2 driver based on existing profile
I1113 17:40:43.693692   11335 start.go:298] selected driver: qemu2
I1113 17:40:43.693705   11335 start.go:902] validating driver "qemu2" against &{Name:minikube KeepContext:false EmbedCerts:false MinikubeISO:https://storage.googleapis.com/minikube/iso/minikube-v1.32.1-arm64.iso KicBaseImage:gcr.io/k8s-minikube/kicbase:v0.0.42@sha256:d35ac07dfda971cabee05e0deca8aeac772f885a5348e1a0c0b0a36db20fcfc0 Memory:8192 CPUs:6 DiskSize:40960 VMDriver: Driver:qemu2 HyperkitVpnKitSock: HyperkitVSockPorts:[] DockerEnv:[] ContainerVolumeMounts:[] InsecureRegistry:[] RegistryMirror:[] HostOnlyCIDR:192.168.59.1/24 HypervVirtualSwitch: HypervUseExternalSwitch:false HypervExternalAdapter: KVMNetwork:default KVMQemuURI:qemu:///system KVMGPU:false KVMHidden:false KVMNUMACount:1 APIServerPort:0 DockerOpt:[] DisableDriverMounts:false NFSShare:[] NFSSharesRoot:/nfsshares UUID: NoVTXCheck:false DNSProxy:false HostDNSResolver:true HostOnlyNicType:virtio NatNicType:virtio SSHIPAddress: SSHUser:root SSHKey: SSHPort:22 KubernetesConfig:{KubernetesVersion:v1.28.3 ClusterName:minikube Namespace:default APIServerName:minikubeCA APIServerNames:[] APIServerIPs:[] DNSDomain:cluster.local ContainerRuntime:docker CRISocket: NetworkPlugin:cni FeatureGates: ServiceCIDR:10.96.0.0/12 ImageRepository: LoadBalancerStartIP: LoadBalancerEndIP: CustomIngressCert: RegistryAliases: ExtraOptions:[] ShouldLoadCachedImages:true EnableDefaultCNI:false CNI:cilium NodeIP: NodePort:8443 NodeName:} Nodes:[{Name: IP: Port:8443 KubernetesVersion:v1.28.3 ContainerRuntime:docker ControlPlane:true Worker:true}] Addons:map[] CustomAddonImages:map[] CustomAddonRegistries:map[] VerifyComponents:map[apiserver:true system_pods:true] StartHostTimeout:6m0s ScheduledStop:<nil> ExposedPorts:[] ListenAddress: Network:socket_vmnet Subnet: MultiNodeRequested:false ExtraDisks:0 CertExpiration:26280h0m0s Mount:true MountString:/Users/nowakb/Work/webp:/web Mount9PVersion:9p2000.L MountGID:docker MountIP: MountMSize:262144 MountOptions:[] MountPort:0 MountType:9p MountUID:docker BinaryMirror: DisableOptimizations:false DisableMetrics:false CustomQemuFirmwarePath: SocketVMnetClientPath:/opt/homebrew/opt/socket_vmnet/bin/socket_vmnet_client SocketVMnetPath:/opt/homebrew/var/run/socket_vmnet StaticIP: SSHAuthSock: SSHAgentPID:0 AutoPauseInterval:1m0s GPUs:}
I1113 17:40:43.693804   11335 start.go:913] status for qemu2: {Installed:true Healthy:true Running:true NeedsImprovement:false Error:<nil> Reason: Fix: Doc: Version:}
I1113 17:40:43.693945   11335 cni.go:84] Creating CNI manager for "cilium"
I1113 17:40:43.693954   11335 start_flags.go:323] config:
{Name:minikube KeepContext:false EmbedCerts:false MinikubeISO:https://storage.googleapis.com/minikube/iso/minikube-v1.32.1-arm64.iso KicBaseImage:gcr.io/k8s-minikube/kicbase:v0.0.42@sha256:d35ac07dfda971cabee05e0deca8aeac772f885a5348e1a0c0b0a36db20fcfc0 Memory:8192 CPUs:6 DiskSize:40960 VMDriver: Driver:qemu2 HyperkitVpnKitSock: HyperkitVSockPorts:[] DockerEnv:[] ContainerVolumeMounts:[] InsecureRegistry:[] RegistryMirror:[] HostOnlyCIDR:192.168.59.1/24 HypervVirtualSwitch: HypervUseExternalSwitch:false HypervExternalAdapter: KVMNetwork:default KVMQemuURI:qemu:///system KVMGPU:false KVMHidden:false KVMNUMACount:1 APIServerPort:0 DockerOpt:[] DisableDriverMounts:false NFSShare:[] NFSSharesRoot:/nfsshares UUID: NoVTXCheck:false DNSProxy:false HostDNSResolver:true HostOnlyNicType:virtio NatNicType:virtio SSHIPAddress: SSHUser:root SSHKey: SSHPort:22 KubernetesConfig:{KubernetesVersion:v1.28.3 ClusterName:minikube Namespace:default APIServerName:minikubeCA APIServerNames:[] APIServerIPs:[] DNSDomain:cluster.local ContainerRuntime:docker CRISocket: NetworkPlugin:cni FeatureGates: ServiceCIDR:10.96.0.0/12 ImageRepository: LoadBalancerStartIP: LoadBalancerEndIP: CustomIngressCert: RegistryAliases: ExtraOptions:[] ShouldLoadCachedImages:true EnableDefaultCNI:false CNI:cilium NodeIP: NodePort:8443 NodeName:} Nodes:[{Name: IP: Port:8443 KubernetesVersion:v1.28.3 ContainerRuntime:docker ControlPlane:true Worker:true}] Addons:map[] CustomAddonImages:map[] CustomAddonRegistries:map[] VerifyComponents:map[apiserver:true system_pods:true] StartHostTimeout:6m0s ScheduledStop:<nil> ExposedPorts:[] ListenAddress: Network:socket_vmnet Subnet: MultiNodeRequested:false ExtraDisks:0 CertExpiration:26280h0m0s Mount:true MountString:/Users/nowakb/Work/webp:/web Mount9PVersion:9p2000.L MountGID:docker MountIP: MountMSize:262144 MountOptions:[] MountPort:0 MountType:9p MountUID:docker BinaryMirror: DisableOptimizations:false DisableMetrics:false CustomQemuFirmwarePath: SocketVMnetClientPath:/opt/homebrew/opt/socket_vmnet/bin/socket_vmnet_client SocketVMnetPath:/opt/homebrew/var/run/socket_vmnet StaticIP: SSHAuthSock: SSHAgentPID:0 AutoPauseInterval:1m0s GPUs:}
I1113 17:40:43.732686   11335 iso.go:125] acquiring lock: {Name:mk33b9e8aa9864dcd98aa574c2ed22eacbc7a850 Clock:{} Delay:500ms Timeout:10m0s Cancel:<nil>}
I1113 17:40:43.736745   11335 out.go:177] 👍  Starting control plane node minikube in cluster minikube
👍  Starting control plane node minikube in cluster minikube
I1113 17:40:43.744742   11335 preload.go:132] Checking if preload exists for k8s version v1.28.3 and runtime docker
I1113 17:40:43.744784   11335 preload.go:148] Found local preload: /Users/nowakb/.minikube/cache/preloaded-tarball/preloaded-images-k8s-v18-v1.28.3-docker-overlay2-arm64.tar.lz4
I1113 17:40:43.744796   11335 cache.go:56] Caching tarball of preloaded images
I1113 17:40:43.744868   11335 preload.go:174] Found /Users/nowakb/.minikube/cache/preloaded-tarball/preloaded-images-k8s-v18-v1.28.3-docker-overlay2-arm64.tar.lz4 in cache, skipping download
I1113 17:40:43.744911   11335 cache.go:59] Finished verifying existence of preloaded tar for  v1.28.3 on docker
I1113 17:40:43.744969   11335 profile.go:148] Saving config to /Users/nowakb/.minikube/profiles/minikube/config.json ...
I1113 17:40:43.745471   11335 start.go:365] acquiring machines lock for minikube: {Name:mke2d79f169af587714a168eb3e0fdf6ff72f286 Clock:{} Delay:500ms Timeout:13m0s Cancel:<nil>}
I1113 17:40:43.745541   11335 start.go:369] acquired machines lock for "minikube" in 50.375µs
I1113 17:40:43.745554   11335 start.go:93] Provisioning new machine with config: &{Name:minikube KeepContext:false EmbedCerts:false MinikubeISO:https://storage.googleapis.com/minikube/iso/minikube-v1.32.1-arm64.iso KicBaseImage:gcr.io/k8s-minikube/kicbase:v0.0.42@sha256:d35ac07dfda971cabee05e0deca8aeac772f885a5348e1a0c0b0a36db20fcfc0 Memory:8192 CPUs:6 DiskSize:40960 VMDriver: Driver:qemu2 HyperkitVpnKitSock: HyperkitVSockPorts:[] DockerEnv:[] ContainerVolumeMounts:[] InsecureRegistry:[] RegistryMirror:[] HostOnlyCIDR:192.168.59.1/24 HypervVirtualSwitch: HypervUseExternalSwitch:false HypervExternalAdapter: KVMNetwork:default KVMQemuURI:qemu:///system KVMGPU:false KVMHidden:false KVMNUMACount:1 APIServerPort:0 DockerOpt:[] DisableDriverMounts:false NFSShare:[] NFSSharesRoot:/nfsshares UUID: NoVTXCheck:false DNSProxy:false HostDNSResolver:true HostOnlyNicType:virtio NatNicType:virtio SSHIPAddress: SSHUser:root SSHKey: SSHPort:22 KubernetesConfig:{KubernetesVersion:v1.28.3 ClusterName:minikube Namespace:default APIServerName:minikubeCA APIServerNames:[] APIServerIPs:[] DNSDomain:cluster.local ContainerRuntime:docker CRISocket: NetworkPlugin:cni FeatureGates: ServiceCIDR:10.96.0.0/12 ImageRepository: LoadBalancerStartIP: LoadBalancerEndIP: CustomIngressCert: RegistryAliases: ExtraOptions:[] ShouldLoadCachedImages:true EnableDefaultCNI:false CNI:cilium NodeIP: NodePort:8443 NodeName:} Nodes:[{Name: IP: Port:8443 KubernetesVersion:v1.28.3 ContainerRuntime:docker ControlPlane:true Worker:true}] Addons:map[] CustomAddonImages:map[] CustomAddonRegistries:map[] VerifyComponents:map[apiserver:true system_pods:true] StartHostTimeout:6m0s ScheduledStop:<nil> ExposedPorts:[] ListenAddress: Network:socket_vmnet Subnet: MultiNodeRequested:false ExtraDisks:0 CertExpiration:26280h0m0s Mount:true MountString:/Users/nowakb/Work/webp:/web Mount9PVersion:9p2000.L MountGID:docker MountIP: MountMSize:262144 MountOptions:[] MountPort:0 MountType:9p MountUID:docker BinaryMirror: DisableOptimizations:false DisableMetrics:false CustomQemuFirmwarePath: SocketVMnetClientPath:/opt/homebrew/opt/socket_vmnet/bin/socket_vmnet_client SocketVMnetPath:/opt/homebrew/var/run/socket_vmnet StaticIP: SSHAuthSock: SSHAgentPID:0 AutoPauseInterval:1m0s GPUs:} &{Name: IP: Port:8443 KubernetesVersion:v1.28.3 ContainerRuntime:docker ControlPlane:true Worker:true}
I1113 17:40:43.745652   11335 start.go:125] createHost starting for "" (driver="qemu2")
I1113 17:40:43.754650   11335 out.go:204] 🔥  Creating qemu2 VM (CPUs=6, Memory=8192MB, Disk=40960MB) ...
🔥  Creating qemu2 VM (CPUs=6, Memory=8192MB, Disk=40960MB) ...| I1113 17:40:43.777521   11335 start.go:159] libmachine.API.Create for "minikube" (driver="qemu2")
I1113 17:40:43.777552   11335 client.go:168] LocalClient.Create starting
I1113 17:40:43.777631   11335 main.go:141] libmachine: Reading certificate data from /Users/nowakb/.minikube/certs/ca.pem
I1113 17:40:43.777676   11335 main.go:141] libmachine: Decoding PEM data...
I1113 17:40:43.777689   11335 main.go:141] libmachine: Parsing certificate...
I1113 17:40:43.777727   11335 main.go:141] libmachine: Reading certificate data from /Users/nowakb/.minikube/certs/cert.pem
I1113 17:40:43.777761   11335 main.go:141] libmachine: Decoding PEM data...
I1113 17:40:43.777769   11335 main.go:141] libmachine: Parsing certificate...
I1113 17:40:43.778137   11335 main.go:141] libmachine: Downloading /Users/nowakb/.minikube/cache/boot2docker.iso from file:///Users/nowakb/.minikube/cache/iso/arm64/minikube-v1.32.1-arm64.iso...
/ I1113 17:40:43.953227   11335 main.go:141] libmachine: Creating SSH key...
\ I1113 17:40:44.079921   11335 main.go:141] libmachine: Creating Disk image...
I1113 17:40:44.079933   11335 main.go:141] libmachine: Creating 40960 MB hard disk image...
I1113 17:40:44.080129   11335 main.go:141] libmachine: executing: qemu-img convert -f raw -O qcow2 /Users/nowakb/.minikube/machines/minikube/disk.qcow2.raw /Users/nowakb/.minikube/machines/minikube/disk.qcow2
I1113 17:40:44.096252   11335 main.go:141] libmachine: STDOUT: 
I1113 17:40:44.096277   11335 main.go:141] libmachine: STDERR: 
I1113 17:40:44.096330   11335 main.go:141] libmachine: executing: qemu-img resize /Users/nowakb/.minikube/machines/minikube/disk.qcow2 +40960M
I1113 17:40:44.110932   11335 main.go:141] libmachine: STDOUT: Image resized.

I1113 17:40:44.110971   11335 main.go:141] libmachine: STDERR: 
I1113 17:40:44.110987   11335 main.go:141] libmachine: DONE writing to /Users/nowakb/.minikube/machines/minikube/disk.qcow2.raw and /Users/nowakb/.minikube/machines/minikube/disk.qcow2
I1113 17:40:44.110996   11335 main.go:141] libmachine: Starting QEMU VM...
I1113 17:40:44.111023   11335 main.go:141] libmachine: executing: /opt/homebrew/opt/socket_vmnet/bin/socket_vmnet_client /opt/homebrew/var/run/socket_vmnet qemu-system-aarch64 -M virt -cpu host -drive file=/opt/homebrew/opt/qemu/share/qemu/edk2-aarch64-code.fd,readonly=on,format=raw,if=pflash -display none -accel hvf -m 8192 -smp 6 -boot d -cdrom /Users/nowakb/.minikube/machines/minikube/boot2docker.iso -qmp unix:/Users/nowakb/.minikube/machines/minikube/monitor,server,nowait -pidfile /Users/nowakb/.minikube/machines/minikube/qemu.pid -device virtio-net-pci,netdev=net0,mac=0e:02:a9:a8:50:58 -netdev socket,id=net0,fd=3 -daemonize /Users/nowakb/.minikube/machines/minikube/disk.qcow2
I1113 17:40:44.118254   11335 main.go:141] libmachine: STDOUT: 
I1113 17:40:44.118290   11335 main.go:141] libmachine: STDERR: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Permission denied

OUTPUT: 
ERROR: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Permission denied

I1113 17:40:44.118311   11335 client.go:171] LocalClient.Create took 340.754833ms
\ I1113 17:40:46.119503   11335 start.go:128] duration metric: createHost completed in 2.373835375s
I1113 17:40:46.119571   11335 start.go:83] releasing machines lock for "minikube", held for 2.374029875s
W1113 17:40:46.119695   11335 start.go:691] error starting host: creating host: create: creating: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Permission denied: exit status 1
I1113 17:40:46.132879   11335 out.go:177] 🔥  Deleting "minikube" in qemu2 ...

🔥  Deleting "minikube" in qemu2 ...
W1113 17:40:46.140657   11335 out.go:239] 🤦  StartHost failed, but will try again: creating host: create: creating: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Permission denied: exit status 1
🤦  StartHost failed, but will try again: creating host: create: creating: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Permission denied: exit status 1
I1113 17:40:46.140690   11335 start.go:706] Will try again in 5 seconds ...
```

### Operating System

macOS (Default)

### Driver

QEMU

**Comment by DuMaM** [comment]
Ok, issue fixed by adding a user to `staff` group which is the default for socket_vemn
`sudo dseditgroup -o edit -a $USER -t user staff`

``` bash
 socket_vmnet --hlep
socket_vmnet: unrecognized option `--hlep'
Usage: socket_vmnet [OPTION]... SOCKET
vmnet.framework support for rootless QEMU.
socket_vmnet does not require QEMU to run as the root user, but socket_vmnet itself has to run as the root, in most cases.

--socket-group=GROUP                socket group name (default: "staff")
--vmnet-mode=(host|shared|bridged)  vmnet mode (default: "shared")
--vmnet-interface=INTERFACE         interface used for --vmnet=bridged, e.g., "en0"
--vmnet-gateway=IP                  gateway used for --vmnet=(host|shared), e.g., "192.168.105.1" (default: decided by macOS)
                                    the next IP (e.g., "192.168.105.2") is used as the first DHCP address
--vmnet-dhcp-end=IP                 end of the DHCP range (default: XXX.XXX.XXX.254)
                                    requires --vmnet-gateway to be specified
--vmnet-mask=MASK                   subnet mask (default: "255.255.255.0")
                                    requires --vmnet-gateway to be specified
--vmnet-interface-id=UUID           vmnet interface ID (default: random)
--vmnet-nat66-prefix=PREFIX::       The IPv6 prefix to use with shared mode.
                                    The prefix must be a ULA i.e. start with fd00::/8.
                                    (default: random)
-p, --pidfile=PIDFILE               save pid to PIDFILE
-h, --help                          display this help and exit
-v, --version                       display version information and exit

version: 1.1.3
```

**Comment by spowelljr** [comment]
So was this an issue related to your socket_vmnet install? Or did the minikube change anything? I can provide a link to minikube v1.31.2 if you are interested in doing any testing. From memory we haven't touched any QEMU code since v1.31.2 so it seems unlikely, but still possible.

**Comment by DuMaM** [comment]
It looks like it was due to a system upgrade which was aligned with minikube update via brew.

**Comment by torenware** [comment]
I've tried @DuMaM's suggested fix using v1.32.0, and it fails:

```
$ minikube start --driver qemu --network socket_vmnet
😄  minikube v1.32.0 on Darwin 14.3 (arm64)
✨  Using the qemu2 driver based on existing profile
👍  Starting control plane node minikube in cluster minikube
🔄  Restarting existing qemu2 VM for "minikube" ...
OUTPUT:
ERROR: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Connection refused

🤦  StartHost failed, but will try again: driver start: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Connection refused: exit status 1
🔄  Restarting existing qemu2 VM for "minikube" ...
OUTPUT:
ERROR: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Connection refused

😿  Failed to start qemu2 VM. Running "minikube delete" may fix it: driver start: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Connection refused: exit status 1

❌  Exiting due to GUEST_PROVISION: error provisioning guest: Failed to start host: driver start: Failed to connect to "/opt/homebrew/var/run/socket_vmnet": Connection refused: exit status 1

╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯

```

**Comment by nikita-fuchs** [comment]
This issue still persists in recent versions ! Why is it closed ?
